coveralls/reporter.py

```diff
@@ -55,20 +56,23 @@ def get_hits(self, line_num, analysis):
     def parse_file(self, cu, analysis):
         """ Generate data for single file """
         filename = cu.file_locator.relative_filename(cu.filename)
-        coverage_lines = [self.get_hits(i, analysis) for i in range(1, len(analysis.parser.lines) + 1)]
-        source_file = cu.source_file()
-        try:
-            source = source_file.read()
-            if sys.version_info < (3, 0):
-                encoding = source_encoding(source)
-                if encoding != 'utf-8':
-                    source = source.decode(encoding).encode('utf-8')
-        except UnicodeDecodeError:
-            log.warn('Source file %s can not be properly decoded, skipping. '
-                     'Please check if encoding declaration is ok', basename(cu.filename))
-            return
-        finally:
-            source_file.close()
+        if __version__.startswith('3.'):
```

Comment:
Agreed. I had actually written it that way at first, I only changed it to match the version sniffing for python. I've updated the diff.